api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/ServiceRequestTranslatorImpl.java

```diff
@@ -37,8 +49,20 @@ public ServiceRequest toFhirResource(TestOrder order) {
 		serviceRequest.setId(order.getUuid());
 		
 		serviceRequest.setStatus(determineServiceRequestStatus(order.getUuid()));
+		
 		serviceRequest.setCode(conceptTranslator.toFhirResource(order.getConcept()));
+		
 		serviceRequest.setIntent(ServiceRequest.ServiceRequestIntent.ORDER);
+		
+		serviceRequest.setSubject(patientReferenceTranslator.toFhirResource(order.getPatient()));
+		
+		serviceRequest.setRequester(providerReferenceTranslator.toFhirResource(order.getOrderer()));
+		
+		serviceRequest.setPerformer(Collections.singletonList(determineServiceRequestPerformer(order.getUuid())));
+		
+		serviceRequest
```

Comment:
Does this work if there is no `effectiveStopDate`? I'm not sure if this parameter makes sense in the context of a one-off non-medication order.

Reply:
Yeah, I wasn't sure how exactly to model the effective dates to dateTime/Period/Timing. 

I'll talk about this at the design forum, but for now I'll set a period like this, and have an empty start and/or end. 

The FHIR docs mention this about missing times in period: 

> If the start element is missing, the start of the period is not known. If the end element is missing, it means that the period is ongoing, or the start may be in the past, and the end date in the future, which means that period is expected/planned to end at the specified time

This seems to generally capture how I understand effective start and end dates to work, but we could also model the occurrence as a single Datetime if there's only an effective start date or only an effective end date. 

(https://www.hl7.org/fhir/servicerequest-definitions.html#ServiceRequest.occurrence_x_)